### PR TITLE
GHI-19: Basic top nav

### PR DESCRIPTION
**Pull request by attiks**
Basic top nav, depends on https://github.com/UN-OCHA/ghi-site/pull/11 and clean up of content types

### CODE REVIEW

File: html/modules/custom/ghi_plans/src/Plugin/ParagraphHandler/PlanTabSwitcher.php [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+    $subform['show_count'] = [
+      '#title' => $this->t('Show count'),
+      '#type' => 'radios',
+      '#default_value' => $config['show_count'],
+      '#options' => [
+        TRUE => $this->t('Yes'),
+        0 => $this->t('No'),
+      ],
+    ];
```
Comment (berliner) [inline html/modules/custom/ghi_plans/src/Plugin/ParagraphHandler/PlanTabSwitcher.php:152]: This should be a setting on the plan node that this paragraph is attached to, either directly or via subpages.
For subpages of type `plan_entity` it should follow the setting in `field_plan_hide_ple_counts`, for subpages of type `governing_entity` it should follow the setting in `field_plan_hide_gve_counts`. We don't want this to be manually set for every single subpage to keep consistency.

File: html/modules/custom/hpc_api/src/Query/EndpointPlanQuery.php [hunk 2]
```diff
@@ -200,4 +208,30 @@ public function getFlowSourcesByPlan($plan_id) {
     return $sources;
   }
 
+  /**
+   * Get plan icon.
+   */
+  public function getPlanIcon($plan_id) {
```
Comment (berliner) [inline html/modules/custom/hpc_api/src/Query/EndpointPlanQuery.php:214]: There are no plan icons in the API. The only icons there are and that Insights cares about are GVE (cluster) icons, see https://hum-insight.info/plan/1037/ge/6030 for an example. But those are already in a sub navigation element, so I think they are not needed here.

File: html/modules/custom/hpc_api/src/Query/EndpointPlanQuery.php [hunk 3]
```diff
@@ -34,10 +34,17 @@ public function getPlanPublic($plan_id) {
...
       'revised_requirements' => $data->revisedRequirements ?? 0,
+      'icon' => '',
       'categories' => [],
       'locations' => [],
     ];
 
+    // Add icon.
+    if (isset($data->governingEntities) && isset($data->governingEntities[0])) {
+      $plan['icon'] = $data->governingEntities[0]->governingEntityVersion->value->icon ?? '';
+    }
```
Comment (berliner) [inline html/modules/custom/hpc_api/src/Query/EndpointPlanQuery.php:45]: See below, there are no plan icons, and fetching the first GVE icon to use as a plan icon is semantically wrong.

File: html/modules/custom/ghi_plans/src/Plugin/ParagraphHandler/PlanTabSwitcher.php [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    $subform['render_all_children'] = [
+      '#title' => $this->t('Render all children'),
+      '#type' => 'radios',
+      '#default_value' => $config['render_all_children'],
+      '#options' => [
+        TRUE => $this->t('Yes'),
+        0 => $this->t('No'),
+      ],
+    ];
+  }
```
Comment (berliner) [inline html/modules/custom/ghi_plans/src/Plugin/ParagraphHandler/PlanTabSwitcher.php:163]: Not quite sure what this is needed for.

File: html/modules/custom/ghi_plans/src/Plugin/ParagraphHandler/PlanTabSwitcher.php [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+    ]);
+
+    if ($plan_entities) {
+      $suffix = '';
+      if ($config['show_count']) {
+        $suffix = ' <span class="counter">' . count($plan_entities) . '</span>';
+      }
+
+      $tabs[] = [
+        'title' => $this->t('Clusters') . $suffix,
```
Comment (berliner) [inline html/modules/custom/ghi_plans/src/Plugin/ParagraphHandler/PlanTabSwitcher.php]: The name of the GVE section can differ from plan to plan, this needs to be retrieved from the GVE prototype. As those prototypes should be the same across all GVEs for a plan, it's ok to use the first one. https://github.com/UN-OCHA/ghi-site/blob/8c0ee84468788445e085a92a516f8a8754e1e098/html/modules/custom/hpc_api/src/Helpers/ApiEntityHelper.php#L280 should give you what's needed then.
Also, `plan entities` are not clusters. `governing_entities` are clusters.

**Review by berliner — COMMENTED** [review]
First review, lot's of things. One main question arises: If this is done as a paragraph handler, how do we prevent editors from moving it around or deleting it? This is a structural navigation item which should be setup in the same way for every plan and plan subpages.
Also, what's your idea for this, should it also cover the non-plan related navigation or should that be separate? In current Insights there is not yet a non-plan navigation in use, though I did prepare something in https://github.com/UN-OCHA/hpc-viewer/blob/943bd4be66b7a14e1d446742fb70faf7ec9d3474/html/sites/all/modules/custom/hpc_content_panes/hpc_content_panes.navigation.inc#L3

**Comment by berliner** [comment]
@attiks And maybe we should change the machine names of those landing page content types. They come from panopoly in D7 and aren't even used in Insight. As there is no more panopoly in in this project, it feels wrong to still reference them.

**Review by attiks — COMMENTED** [review]
> And maybe we should change the machine names of those landing page content types

@berliner can you take care of this?

File: html/modules/custom/ghi_plans/src/Plugin/ParagraphHandler/PlanTabSwitcher.php (hunk 5 above)
Comment (attiks) [reply html/modules/custom/ghi_plans/src/Plugin/ParagraphHandler/PlanTabSwitcher.php]: I assumed all content types are in use

File: html/modules/custom/hpc_api/src/Query/EndpointPlanQuery.php [hunk 6]
```diff
@@ -276,4 +284,30 @@ public function getFlowSourcesByPlan($plan_id) {
...
+
+    $icon = $plan->icon;
+    if (empty($icon)) {
+      return;
+    }
+
+    $this->setEndpoint('v2/icon/' . $icon);
+    $svg_data = $this->getData();
+    $svg_content = $svg_data ? $svg_data->svg : NULL;
```
[the remark this answers is not in the document]
Comment (berliner) [reply html/modules/custom/hpc_api/src/Query/EndpointPlanQuery.php:228]: No, should be something like this:
```
$ hpc "https://api.hpc.tools/v2/icon/clusters_camp_coordination_and_camp_management_icon"
{
  "data": {
    "id": "clusters_camp_coordination_and_camp_management_icon",
    "svg": "<svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 48 44\"><defs><style>.cls-1{fill:#5091cd;}</style></defs><g id=\"Layer_2\" data-name=\"Layer 2\"><g id=\"Icons\"><path class=\"cls-1\" d=\"M4.49,17.6,24,6.45,43.51,17.6A3,3,0,0,0,45,18a3,3,0,0,0,1.49-5.6l-21-12a3,3,0,0,0-3,0l-21,12a3,3,0,0,0,3,5.21Z\"/><path class=\"cls-1\" d=\"M38,27a9.91,9.91,0,0,0-8.65,4.59,6.71,6.71,0,0,0-10.7,0A9.91,9.91,0,0,0,10,27C4.3,27,0,31.09,0,36.48a1.5,1.5,0,0,0,3,0,6.21,6.21,0,0,1,3-5.36V40a4,4,0,0,0,8,0V31.12a6.21,6.21,0,0,1,3,5.36,1.5,1.5,0,0,0,3,0v0h0a4.8,4.8,0,0,1,1-2.94V37a3,3,0,0,0,6,0V33.5a4.75,4.75,0,0,1,1,2.94s0,0,0,0a1.5,1.5,0,0,0,3,0,6.24,6.24,0,0,1,3.6-5.69L33,38a5.09,5.09,0,1,0,10,0L41.4,30.79A6.24,6.24,0,0,1,45,36.48a1.5,1.5,0,0,0,3,0C48,31.09,43.7,27,38,27Z\"/><circle class=\"cls-1\" cx=\"38\" cy=\"22\" r=\"3\"/><circle class=\"cls-1\" cx=\"10\" cy=\"22\" r=\"3\"/><path class=\"cls-1\" d=\"M21.5,24.49A2.51,2.51,0,1,0,24,22,2.5,2.5,0,0,0,21.5,24.49Z\"/></g></g></svg>"
  }
}
```

Where did you get that?

**Comment by berliner** [comment]
@attiks Do we need those landing page content types? What's your idea of using them?

**Comment by attiks** [comment]
> @attiks Do we need those landing page content types? What's your idea of using them?

No idea, I assumed since they are there we would need them, can you remove everything we don't need (for now)

**Comment by attiks** [comment]
> This is a structural navigation item which should be setup in the same way for every plan and plan subpage

I'll remove the paragraph and move the logic to the node preprocess so it's available to all plans and gov entities

**Pull request merged**